1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,91 @@
-function setAlarm() {}
+// Access to input
+const alermSetInput = document.querySelector("#alarmSet");
+// Access to span for showing timer
+const timeRemaining = document.querySelector(".timeRange");
+
+let intervalID;
+let intervalColor;
+
+function setAlarm() {
+  if (alermSetInput.value === "") {
+    alert("Please enter the proper input");
+  } else {
+    // Turn string value to number
+    let totalSeconds = alermSetInput.value;
+    // convert sec to 00:00 form
+    let timer = convertSecondsToMinutesAndSeconds(totalSeconds);
+    // Show timer in time remaining
+    timeRemaining.textContent = timer;
+    document.body.style.backgroundColor = "green";
+
+    // Set interval for timer
+    intervalID = setInterval(() => {
+      // Decrease time by 1 sec
+      timer = decrementByOneSec(timer);
+      if (timer === "00:00") {
+        // timeRemaining.textContent = timer;
+        document.body.style.backgroundColor = "red";
+        // Deactive interval
+        clearInterval(intervalID);
+        // Active alarm
+        playAlarm();
+        // stop alarm after 100 s
+        setTimeout(() => {
+          stopAlerm();
+        }, 100000);
+        intervalColor = setInterval(() => {
+          if (document.body.style.backgroundColor === "red") {
+            document.body.style.backgroundColor = "white";
+          } else {
+            document.body.style.backgroundColor = "red";
+          }
+        }, 500);
+      }
+    }, 1000);
+  }
+}
+
+// Convert number to a form of 00:00
+let convertSecondsToMinutesAndSeconds = function (totalSeconds) {
```

Comment:
Did you intentionally use a function expression here, even though some of the functions in this file are function declared using function declarations? While there's nothing inherently wrong with using function expressions, it's important to keep in mind that they are not hoisted. To avoid potential bugs, I recommend consistently using either function expressions or function declarations throughout your codebase, unless you have a specific reason to do otherwise.

Reply:
Yes, I used it with no reason and I changed it to a function declarations.